src/common/libschedutil/init.c

```diff
@@ -0,0 +1,111 @@
+/************************************************************\
+ * Copyright 2019 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#include <czmq.h>
+#include <errno.h>
+#include <flux/core.h>
+
+#include "src/common/libutil/aux.h"
+
+#include "init.h"
+
+struct schedutil_ctx {
+    flux_t *h;
+    struct aux_item *aux;
+    bool test_pause; // used for testing that schedutil responds to outstanding futures when sched is unloaded
+    zlistx_t *outstanding_msgs;
+};
+
+schedutil_t *schedutil_create (flux_t *h)
+{
+    schedutil_t *s = calloc(1, sizeof(*s));
+    if (!s)
+        return NULL;
+
+    s->h = h;
+    s->test_pause = false;
+    s->outstanding_msgs = zlistx_new ();
+
+    return s;
+}
+
+static void respond_to_outstanding_msgs (schedutil_t *s)
+{
+    int rc = 0;
+    flux_msg_t *msg;
+    for (msg = zlistx_first (s->outstanding_msgs);
+         msg;
+         msg = zlistx_next (s->outstanding_msgs))
+        {
+            rc += flux_respond_error (s->h, msg, ENOSYS, "automatic ENOSYS response from schedutil");
```

Comment:
style: This line is too long (shame, shame), and the curly brace for this block should be on the same line as the `for` loop final paren. :smile:

Reply:
Fixed along with some other abuses in https://github.com/flux-framework/flux-core/pull/2226/commits/263a28b0c8b410ca50c8e3a75218526dacd79198